src/flow/src/fe_client.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Frontend Client for flownode, used for writing result back to database
+
+use api::v1::greptime_database_client::GreptimeDatabaseClient;
+use api::v1::greptime_request::Request;
+use api::v1::{
+    GreptimeRequest, GreptimeResponse, RequestHeader, RowDeleteRequests, RowInsertRequests,
+};
+use common_error::ext::BoxedError;
+use common_frontend::handler::FrontendInvoker;
+use common_query::Output;
+use common_telemetry::tracing_context::W3cTrace;
+use session::context::{QueryContext, QueryContextRef};
+use snafu::IntoError;
+use tokio::sync::Mutex;
+
+use crate::{Error, Result};
+
+/// Frontend client for writing result back to database
+pub struct FrontendClient {
+    client: GreptimeDatabaseClient<tonic::transport::Channel>,
+}
+
+impl FrontendClient {
+    pub fn new(channel: tonic::transport::Channel) -> Self {
+        Self {
+            client: GreptimeDatabaseClient::new(channel),
+        }
+    }
+}
+
+fn to_rpc_request(request: Request, ctx: &QueryContext) -> GreptimeRequest {
+    let header = RequestHeader {
+        catalog: ctx.current_catalog().to_string(),
+        schema: ctx.current_schema().to_string(),
+        authorization: None,
+        // dbname is empty so that header use catalog+schema to determine the database
+        // see `create_query_context` in `greptime_handler.rs`
+        dbname: "".to_string(),
+        timezone: ctx.timezone().to_string(),
+        tracing_context: W3cTrace::new(),
```

Comment:
How about to keep the tracing context
```suggestion
        tracing_context: TracingContext::from_current_span().to_w3c(),
```